messybrainz/data.py

```diff
@@ -241,3 +241,31 @@ def link_recording_to_recording_id(connection, msid, mbid):
         "mbid": mbid,
     })
 
+
+def _convert_to_messybrainz_json(data):
+    """ Converts the specified data dict into JSON strings, while
+    applying MessyBrainz' transformations which include
+        * sorting by keys
+        * lowercasing all values
+
+    Args:
+        data (dict): the dict to be converted into MessyBrainz JSON
+
+    Returns:
+        str: the MessyBrainz JSON after transformations
+    """
+
+    def lower(d):
+        """ Lower cases all values in the specified dict.
+        """
+        result = {}
+        for key, value in d.items():
+            if isinstance(value, str):
+                result[key] = value.lower()
+            elif isinstace(value, dict):
+                result[key] = lower(value)
```

Comment:
If you're doing these list/dict checks then it'd be good to extend the test to include these things too